splunk-otel-android/src/main/java/com/splunk/rum/LogToSpanBridge.java

```diff
@@ -0,0 +1,121 @@
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.logs.Severity;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.api.trace.TracerBuilder;
+import io.opentelemetry.api.trace.TracerProvider;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
+import io.opentelemetry.sdk.logs.LogRecordProcessor;
+import io.opentelemetry.sdk.logs.ReadWriteLogRecord;
+import io.opentelemetry.sdk.logs.data.Body;
+import io.opentelemetry.sdk.logs.data.LogRecordData;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.concurrent.TimeUnit;
+
+final class LogToSpanBridge implements LogRecordProcessor {
+
+    // can be used in Splunk code to set span names in instrumentations
+    static final AttributeKey<String> OPERATION_NAME = stringKey("log.operation_name");
+
+    static final AttributeKey<Long> LOG_SEVERITY = longKey("log.severity");
+    static final AttributeKey<String> LOG_SEVERITY_TEXT = stringKey("log.severity_text");
+    static final AttributeKey<String> LOG_BODY = stringKey("log.body");
+
+    private volatile TracerProvider tracerProvider;
+
+    void setTracerProvider(TracerProvider tracerProvider) {
+        this.tracerProvider = tracerProvider;
+    }
+
+    @Override
+    public void onEmit(Context context, ReadWriteLogRecord logRecord) {
+        TracerProvider tracerProvider = this.tracerProvider;
+        if (tracerProvider == null) {
+            // if this is null then we've messed up the RumInitializer implementation
+            return;
+        }
+
+        LogRecordData log = logRecord.toLogRecordData();
+        Tracer tracer = getTracer(tracerProvider, log.getInstrumentationScopeInfo());
+
+        SpanBuilder spanBuilder = tracer.spanBuilder(getSpanName(log));
+        setLogAttributes(spanBuilder, log);
+        Span span =
+                spanBuilder
+                        .setStartTimestamp(log.getEpochNanos(), TimeUnit.NANOSECONDS)
+                        .startSpan();
+        span.end(log.getEpochNanos(), TimeUnit.NANOSECONDS);
+    }
+
+    private static Tracer getTracer(TracerProvider tracerProvider, InstrumentationScopeInfo scope) {
+        TracerBuilder builder = tracerProvider.tracerBuilder(scope.getName());
+        String version = scope.getVersion();
+        if (version != null) {
+            builder.setInstrumentationVersion(version);
+        }
+        String schemaUrl = scope.getSchemaUrl();
+        if (schemaUrl != null) {
+            builder.setSchemaUrl(schemaUrl);
+        }
+        return builder.build();
+    }
+
+    private static String getSpanName(LogRecordData log) {
+        String operationName = log.getAttributes().get(OPERATION_NAME);
+        if (operationName != null) {
+            return operationName;
+        }
```

Comment:
Spans have to have a name; log records don't have one. In future instrumentations that use logs/events we could use that attribute to inject the span name; e.g. perhaps by using an `AttributesExtractor`-like thing for customization.